app/models/classification.rb

```diff
@@ -17,26 +14,32 @@ class Classification < ActiveRecord::Base
   validate :metadata, :required_metadata_present
   validate :validate_gold_standard
 
-  can :show, :in_show_scope?
-  can :update, :created_and_incomplete?
-  can :destroy, :created_and_incomplete?
-
-  def self.visible_to(actor, as_admin: false)
-    ClassificationVisibilityQuery.new(actor, self).build(as_admin)
-  end
-
-  def self.can_create?(actor)
-    true
+  scope :incomplete, -> { where(completed: false) }
+  scope :created_by, -> (user) { where(user: user) }
+  scope :complete, -> { where(completed: true) }
+
+  def self.scope_for(action, actor, opts={})
+    case action
+    when :show, :index
+      query = joins(:project).merge(Project.scope_for(:update, actor))
+        .union_all(joins(:user_group).merge(actor.user_groups))
+        .union_all(actor.classifications)
+      # Workaround Broken Bind Value Assignment in Subqueries in Rails 4.1
+      # This is fixed in Rails 4.2 when we're able to to migrate to that
```

Comment:
When Rails 4.2.1 is out we should be good https://github.com/rails/rails/issues/18414